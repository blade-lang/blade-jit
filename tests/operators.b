echo 2 + 2
echo 2 * 2
echo ~2
echo 2 >>> -1


class A {
  @new(val) {
    self.val = val
  }

  def + {
    return new A(self.val + __arg__.val)
  }

  def >>> {
    return new A(self.val * __arg__.val)
  }

  def ~ {
    return new A(self.val ** 2)
  }
}

echo (new A(5) + new A(11)).val
echo (new A(5) >>> new A(12)).val
echo (~(~(new A(12)))).val

var g = new A(24)
g += new A(93)

echo g.val

echo ~12
