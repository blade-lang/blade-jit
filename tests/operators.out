4
4
-3
0
16
60
20736
117
-13